### ISSUE TITLE
Emerging good practices for Open Broadcaster System (OBS)

### ISSUE THREAD

**Issue by ragamouf** [carpentries/community-facilitators-program#26]
Share observations (pros and cons) between OBS and Zoom beta for slides or live coding in the background.

Recently I had an interesting teaching demo experience recently where a demonstrator didn't share the screen - instead, used a gaming service application (OBS) to stream the live coding and lesson materials so that these appeared in their zoom background, negating the use of "share screen". They had a greenscreen set up so that their talking head appeared in a lower corner of the screen, maintaining eye contact and producing a virtual effect much more similar to f2f instructing in front of a whiteboard. This has a couple implications for teaching demos. 

Firstly, the obvious benefits of a more accessible presentation for learners who may be lip reading, better arrangement of screen real estate for learners with 1 screen, and a more engaging presentation overall. However, because zoom streaming quality defaults to the user, if a learner doesn't have a stable connection, the quality of the streaming video degrades, and the live coding screen is harder to see. I'm excited by these new possibilities, even though they challenge the current rubric for assessing teaching demonstrations which assumes and recommends seperate screens. 

Here's an example of @chendaniely doing this for [teaching RStudio in ds4biomed](https://youtu.be/6y8qQ_zw-r0?t=340), including instructions for learners to pin his video in zoom, embedded as part of 'how to set up your quadrants if you're on a single monitor'. Note however that his talking head in this example is adjacent and not overlaid as in the first example described.

[Link to similar discussion in the trainers slack channel](https://swcarpentry.slack.com/archives/G7A6ED1SA/p1614901316021200) (note: restricted to trainers)

Recommendations for next steps (happy for any or indeed all to be attempted!)

- [ ] Test an OBS set up & gather evidence by recording in zoom (instructor view) and taking screenshots (both instructor and learner views)
- [ ] Raise for discussion / Q&A demo in a themed Carpentries discussion for instructors later this year
- [ ] Develop guidance for trainers running teaching demos
- [ ] Long term goal = create a bonus module for instructors on how to set up OBS

Any comments or feedback gratefully accepted!

**Comment by bkmgit** [comment]
This is nice. The requirements on the computer streaming OBS are 
relatively high.  BigBlueButton ( https://bigbluebutton.org/ ) also 
works well. Can be self hosted and integrates an Etherpad.

**Comment by annajiat** [comment]
I would agree with @ragamouf that it might be helpful for lip readers, however, I also agree with @bkmgit that it creates issues for:

- instructors without powerful hardware 
- usually screens don't change much for a while in a presentation. However, when video and presentation screen are superimposed, it might increase the bandwidth requirement for both the instructor as well as the participant because at least the presenter video part is changing frequently (@FPS). We have already seen that some participant's experience improves if everyone except the instructor turns off their video. Audio becomes less choppy/less delayed. Adding more payload to the video transmission bandwidth requirement might lead to less diversity/inclusion/accessibility for some of the learners as well as instructors.
- instructors who are unable to share video, may be no camera in desktop
- instructors who are unwilling to share video: family/home background. Virtual backgrounds has bugs.

Instead,
I would like to vote for captioning/transcription services be it manual or automatic.

**Comment by bkmgit** [comment]
There seem to be efforts to integrate this into BigBlueButton:
https://github.com/bigbluebutton/bigbluebutton/issues/9309
There are a number of speech to text engines and technologies, though 
technical terms and speech in multiple languages may be problematic. 
Also discussed on the above thread is the WebSpeechAPI 
https://developer.mozilla.org/en-US/docs/Web/API/Web_Speech_API that can 
do speech to text

**Comment by ragamouf** [comment]
Thanks @annajiat and @bkmgit  for so clearly outlining why loading up the video payload is a problem for accessibility. I would love to follow these links!

I'm glad you brought up captioning, as I was going to create a separate issue on web captioning since I am interested in the social dimensions of how auto captioning can be practically useful, rather than 'looking the part'. For example, using auto captioning services require learning how to use them, and getting used to changing the way you speak so that the captions reflect your words.

**Comment by maneesha** [comment]
This was assigned to me -- is there something specific needed from me now?
